ee/tabby-webserver/src/service/event_logger.rs

```diff
@@ -21,16 +23,35 @@ pub fn create_event_logger(db: DbConn) -> impl EventLogger + 'static {
     DbEventLogger { db }
 }
 
+fn get_user_id(user: Option<String>) -> Option<i64> {
+    user.and_then(|id| ID::new(id).as_rowid().ok())
+}
+
 impl EventLogger for DbEventLogger {
     fn write(&self, entry: LogEntry) {
```

Comment:
i feel this function could benefit with a refactoring adding an `write_impl(...) -> Result` for easier `?` chaining syntax to reduce some verbosity.

feel free to do this as a follow up by adding a FIXME

Reply:
https://linear.app/tabbyml/issue/TAB-629/refactor-webserverserviceevent-loggerrs-write-impl